core/src/models/Mapping.ts

```diff
@@ -49,6 +49,9 @@ export class Mapping extends CommonModel<Mapping> {
   @BelongsTo(() => Property)
   property: Property;
 
+  // TODO: remoteKey can come into play here
+  uniqueIdentifier = ["ownerId", "ownerType", "propertyId"];
```

Comment:
Adding to my comment in `core/src/models/DestinationGroupMembership.ts`
This had `ownerId + ownerType + propertyId` and `ownerId + ownerType + remoteKey` as separate identifiers and combining them caused some tests to fail. Leaning more towards a `[[],[]]` style but hoping not to as the error messages will get even more obnoxious.

Reply:
I think the reason for both of the checks was to prevent both of the following:

1. 
ownerType: property
ownerId: email
remotekey: email_column

^ So you can't have 2 properties for the same source column

2. 
ownerType: destination
ownerId: mailchimp
propertyId: email

^ So you can't export the email property to 2 different mailchimp fieilds.

I think your solution doesn't cover both cases (the first is missing at the moment)... BUT, good news, we want to [change how mappings work rather significantly](https://www.pivotaltracker.com/story/show/181244601) in the near future, and we would need to remove check (2) anyway. 

I think that it's OK to have only one check, but I'd suggest the remoteKey version.

```suggestion
  uniqueIdentifier = ["ownerId", "ownerType", "remoteKey"];
```